lib/puppet/resource/catalog.rb

```diff
@@ -214,6 +219,7 @@ def alias(resource, key)
   # @api public
   def apply(options = {})
     Puppet::Util::Storage.load if host_config?
+    persistence.load if host_config?
```

Comment:
I see why we're loading the persistence file here, but I don't like it. We're adding in the concept of persistence to the catalog which doesn't need to know about it; it just happens that `Puppet::Resource::Catalog#apply` happens to manage the lifetime of the transaction (and knows that we're a host catalog) and the `Storage` class was loaded and saved here, but really neither of these are related to the catalog. I would rather have this moved to `Puppet::Transaction#evaluate` to manage the persistence object lifetime; at that point `Catalog#host_config?` should still be accessible and we won't have to have the persistence file managed so far way from where it's used.